### PR TITLE
Update crossfeed TXT records for acme challenge

### PR DESCRIPTION
**Pull request by lfilip-ivaal**
#  #

## 🗣 Description ##

We need to update the TXT record for the DNS to create a certificate for Let'sEncrypt.  terraform-docs has been run and didn't indicate any changes on my machine, but might still need to be checked

## 💭 Motivation and context ##

Creating this for integration into a new environment.   This will be the first one that we need. Crossfeed will need more going forward.  This is a proof of concept to make sure we can create these effectively.

## 🧪 Testing ##

Ran terraform-docs to see if any changes were made to the docs. None on my machine.
@dav3r applied these changes in Production and verified that they applied cleanly.

## ✅ Pre-approval checklist ##

- [X] This PR has an informative and human-readable title.
- [X] Changes are limited to a single goal - *eschew scope creep!*
- [X] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [X] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [X] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.

### CODE REVIEW

**Comment by lfilip-ivaal** [comment]
We may be having to do a few of these. This is just the proof of concept. Thanks in advance from all of us over here. :). I have tried to do better about naming/labeling commits and titles.

File: route53_crossfeed_app.tf [hunk 1]
```diff
@@ -295,6 +295,18 @@ resource "aws_route53_record" "crossfeed_staging_TXT" {
   zone_id = aws_route53_zone.cyber_dhs_gov.zone_id
 }
 
+resource "aws_route53_record" "crossfeed_staging_ACME_TXT" {
```
Comment (dav3r) [inline route53_crossfeed_app.tf]: I think this is more in line with the resource naming scheme we have used in this file:
```suggestion
resource "aws_route53_record" "crossfeed_staging_acme_TXT" {
```

File: route53_crossfeed_app.tf [hunk 2]
```diff
@@ -281,7 +281,7 @@ resource "aws_route53_record" "crossfeed_staging_MX" {
   zone_id = aws_route53_zone.cyber_dhs_gov.zone_id
 }
 
-resource "aws_route53_record" "crossfeed_staging_TXT" {
+resource "aws_route53_record" "crossfeed_staging_TXT1" {
```
Comment (dav3r) [inline route53_crossfeed_app.tf]: I think this should be modified to align with the resource naming scheme we have used in this file:
```suggestion
resource "aws_route53_record" "crossfeed_staging_ses_TXT" {
```

**Review by dav3r — CHANGES_REQUESTED** [review]
These resource name changes will affect the output of `terraform-docs`.  Did you run this command (from the root directory of this repo)?
```
terraform-docs markdown --anchor=false --html=false --lockfile=false .
```

File: route53_crossfeed_app.tf (hunk 2 above)
Comment (dav3r) [reply route53_crossfeed_app.tf]: My bad @lfilip-ivaal - I didn't notice that we already have a `crossfeed_staging_ses_TXT` in this file.  I think that this resource name can stay the same as it was before:
```
resource "aws_route53_record" "crossfeed_staging_TXT" {
```

See https://github.com/cisagov/cool-dns-cyber.dhs.gov/pull/80#discussion_r1288556217.

File: route53_crossfeed_app.tf [hunk 3]
```diff
@@ -281,7 +281,7 @@ resource "aws_route53_record" "crossfeed_staging_MX" {
   zone_id = aws_route53_zone.cyber_dhs_gov.zone_id
 }
 
-resource "aws_route53_record" "crossfeed_staging_TXT" {
+resource "aws_route53_record" "crossfeed_staging_ses_TXT" {
```
Comment (dav3r) [inline route53_crossfeed_app.tf]: See https://github.com/cisagov/cool-dns-cyber.dhs.gov/pull/80#discussion_r1288554962:
```suggestion
resource "aws_route53_record" "crossfeed_staging_TXT" {
```

File: route53_crossfeed_app.tf (hunk 2 above)
Comment (dav3r) [reply route53_crossfeed_app.tf]: Resolved with https://github.com/cisagov/cool-dns-cyber.dhs.gov/pull/80/commits/dfe350d7cf8a4f5423162c02cc356f6bc9560b93.

**Comment by dav3r** [comment]
> These resource name changes will affect the output of `terraform-docs`. Did you run this command (from the root directory of this repo)?
> 
> ```
> terraform-docs markdown --anchor=false --html=false --lockfile=false .
> ```

@lfilip-ivaal Are you able to get `terraform-docs` to run with the command above?  You should see that an additional line needs to be added for `aws_route53_record.crossfeed_staging_acme_TXT`.

**Comment by lfilip-ivaal** [comment]
> > These resource name changes will affect the output of `terraform-docs`. Did you run this command (from the root directory of this repo)?
> > ```
> > terraform-docs markdown --anchor=false --html=false --lockfile=false .
> > ```
> 
> @lfilip-ivaal Are you able to get `terraform-docs` to run with the command above? You should see that an additional line needs to be added for `aws_route53_record.crossfeed_staging_acme_TXT`.

@dav3r Yes. I was under the impression that running the terraform-docs would automatically update the correct file.  I have now put the changes into the README.md.

File: README.md [hunk 4]
```diff
@@ -79,9 +79,10 @@ zone.  This role has a trust relationship with the users account.
...
 | [aws_route53_record.crossfeed_staging_A](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_record) | resource |
 | [aws_route53_record.crossfeed_staging_AAAA](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_record) | resource |
+| [aws_route53_record.crossfeed_staging_ACME_TXT](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_record) | resource |
 | [aws_route53_record.crossfeed_staging_CNAME](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_record) | resource |
 | [aws_route53_record.crossfeed_staging_MX](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_record) | resource |
-| [aws_route53_record.crossfeed_staging_TXT](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_record) | resource |
+| [aws_route53_record.crossfeed_staging_TXT1](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_record) | resource |
```
Comment (dav3r) [inline README.md]: This change should be reverted to match the change in https://github.com/cisagov/cool-dns-cyber.dhs.gov/pull/80/commits/dfe350d7cf8a4f5423162c02cc356f6bc9560b93:
```suggestion
| [aws_route53_record.crossfeed_staging_TXT](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_record) | resource |
```

**Review by dav3r — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (dav3r) [reply README.md]: Resolved in https://github.com/cisagov/cool-dns-cyber.dhs.gov/pull/80/commits/70f7417059899955ae75429316fa1bd73e9dc148.

**Comment by dav3r** [comment]
Looks good now.  I'm going to attempt to apply this change now and if it works, I'll approve this PR.

**Review by dav3r — APPROVED** [review]
I was able to apply this change successfully - you should see the new TXT record out there shortly.

**Comment by lfilip-ivaal** [comment]
@dav3r  That worked! Ok, I have updated the ticket to remove the previous TXT and added a new TXT challenge to the PR for the other cert we needed to create. 


I should have asked if you guys would have preferred a different PR for that.  Let me know if this works and I can comply with your team's methods going forward. 😄 

Thanks for doing this. Much appreciated.  I think this should be it for now.

**Comment by dav3r** [comment]
I want to make sure I'm understanding what you want here.  After I applied the initial TXT record for `aws_route53_record.crossfeed_staging_acme_TXT`, you removed that record from this PR and added a new one for `aws_route53_record.crossfeed_staging_api_acme_TXT`.  If I reapply these changes, it will remove the initial challenge record (`crossfeed_staging_acme_TXT`) and add a new one for `crossfeed_staging_api_acme_TXT`.  Can you please confirm that is what you want done here?

In the future, I suggest keeping the PR as a [draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) and communicating with us about when you want the changes applied.  Once everything looks correct (i.e. no more changes will be done in the PR), then you should take the PR out of draft mode and request approval.

**Comment by lfilip-ivaal** [comment]
> I want to make sure I'm understanding what you want here. After I applied the initial TXT record for `aws_route53_record.crossfeed_staging_acme_TXT`, you removed that record from this PR and added a new one for `aws_route53_record.crossfeed_staging_api_acme_TXT`. If I reapply these changes, it will remove the initial challenge record (`crossfeed_staging_acme_TXT`) and add a new one for `crossfeed_staging_api_acme_TXT`. Can you please confirm that is what you want done here?
> 
> In the future, I suggest keeping the PR as a [draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) and communicating with us about when you want the changes applied. Once everything looks correct (i.e. no more changes will be done in the PR), then you should take the PR out of draft mode and request approval.


@dav3r Yes, that is exactly what we need to do. Remove the old one and put the new one in there.  Since I was able to create the certificate, that acme challenge content code is no longer any use to anyone, so I didn't want to clutter up your codebase with temporary resources. So even after this resource is in there, we don't need it again after we get the certificate.  

We are doing some things we normally don't and learning as we go.  In the future I will follow the path you suggest. Now that we know how to do it, any future issues can most likely be handled in bulk instead of one at a time.

**Comment by dav3r** [comment]
Alright, `aws_route53_record.crossfeed_staging_acme_TXT` has been removed and `aws_route53_record.crossfeed_staging_api_acme_TXT` has been successfully added.

Are you planning to add/remove any more TXT records in this PR or will those be handled via a future PR?

**Comment by lfilip-ivaal** [comment]
> Alright, `aws_route53_record.crossfeed_staging_acme_TXT` has been removed and `aws_route53_record.crossfeed_staging_api_acme_TXT` has been successfully added.
> 
> Are you planning to add/remove any more TXT records in this PR or will those be handled via a future PR?

We have all the certificates completed now. We don't need those records anymore and they are ok to delete.   Any future certs we will be in a different PR.

Thanks again for all the help. Just let me know if I need to delete/close this PR.

**Comment by dav3r** [comment]
There's no need to delete these TXT records- it doesn't hurt to leave them out there.  They can simply be updated in a new PR the next time they need to change.

I'm going to approve this PR and we will merge once we get another approval from my team.

**Review by dav3r — APPROVED** [review]

**Review by mcdonnnj — APPROVED** [review]

File: route53_crossfeed_app.tf [hunk 5]
```diff
@@ -374,6 +374,18 @@ resource "aws_route53_record" "crossfeed_staging_cd_api_CNAME" {
   zone_id = aws_route53_zone.cyber_dhs_gov.zone_id
 }
 
+resource "aws_route53_record" "crossfeed_staging_api_acme_TXT" {
+  provider = aws.route53resourcechange
+
+  name = "_acme-challenge.api.staging.crossfeed.${aws_route53_zone.cyber_dhs_gov.name}"
```
Comment (epicfaace) [inline route53_crossfeed_app.tf:380]: @lfilip-ivaal did we also need a TXT record for staging.crossfeed? Or only needed one for api.staging.crossfeed?

**Pull request merged**